In [1]:
import numpy as np
import pandas as pd
import pylab as pl
%pylab inline
import scipy.stats as st

Populating the interactive namespace from numpy and matplotlib


/opt/rh/anaconda/root/lib/python2.7/site-packages/pandas/computation/__init__.py:19: UserWarning: The installed version of numexpr 2.4.4 is not supported in pandas and will be not be used

  UserWarning)


## NULL HYPOTHESIS:

#### The % of former inmate was convicted as a felony after release is the same or lower for candidates who participated in the program as for the control group, significance level p=0.05
#### $H_0: P_0 - P_1 \geq 0$ 
#### $H_a: P_0 - P_1 \leq 0$ 
#### $P_0 = 0.117, P_1= 0.1$

In [2]:
alpha=0.05

P_0 = 11.7 * 0.01 
P_1 = 10 * 0.01

if P_0-P_1 >= 0:
    
    print ("the Null holds")
else:
    print ("we must assess the statistical significance")

n_0 = 409
n_1 = 568

Nt_0 = P_0 * n_0
Nt_1 = P_1 * n_1

the Null holds


In [3]:
sp = (P_0 * n_0 + P_1 * n_1) / (n_1 + n_0)
print (sp)

0.107116683726


In [4]:
p = lambda p0, p1, n0, n1: (p0 * n0 + p1 * n1) / (n0 + n1)
se = lambda p, n0, n1: np.sqrt(p * (1 - p) * (1.0 / n0 + 1.0 / n1))

In [5]:
zscore = lambda p0, p1, s : (p0 - p1) / s
z_2y = zscore(P_1, P_0, se(p(P_0, P_1, n_0, n_1), n_0, n_1))
p_2y = st.norm.sf(abs(z_2y)) ## or look up the table
print abs(z_2y)
print p_2y

0.84764020522
0.198319188134


In [6]:
def report_result(p,a):
    print ('is the p value ' + 
           '{0:.2f} smaller than the critical value {1:.2f}?'.format(p,a))
    if p < a:
        print ("YES!")
    else: 
        print ("NO!")
    
    print ('the Null hypothesis is {}'.format(\
                            'rejected' if p < a  else 'not rejected') )

    
report_result(p_2y, alpha)

is the p value 0.20 smaller than the critical value 0.05?
NO!
the Null hypothesis is not rejected


## Fill the table.
|convicted of a fellony | yes | no | 
|:---------------------|----:|:---|
| test sample | 0.100*568 | 0.900*568 | 568 | 
| control sample | 0.117*409 | 0.883*409 | 409| 
|||||
| total | 104.653 | 872.347 | 977|


In [9]:
def evalChisq(values):
    values = np.array(values)
    E = np.empty_like(values)
    for j in range(len(values[0])):
        for i in range(2):
            
            E[i][j] = ((values[i,:].sum() * values[:,j].sum()) / 
                        (values).sum())
    return ((values - E)**2 / E).sum()

In [11]:
Ntot = 997
sample_values = np.array([[0.10 * 568, 0.90 * 568], [0.117 * 409, 0.883 * 409]])
print (evalChisq(sample_values))

0.718493917505


#### When DOF = 1 and  alpha = 0.05, the critical value is 3.841. However, 0.718 is less than 3.841 so we can not reject the Null Hypothesis.